tests/sentry/toolbar/utils/test_csp.py

```diff
@@ -0,0 +1,54 @@
+from sentry.toolbar.utils.csp import csp_add_directive
+
+
+def get_directives(csp: str) -> dict[str, set[str]]:
+    res = {}
+    for directive in csp.split(";"):
+        directive = directive.strip()
+        key, *values = directive.split()
+        res[key] = set(values)
+    return res
+
+
+def run_test(csp: str, new_key: str, new_values: list[str]) -> dict[str, set[str]]:
+    new_csp = csp_add_directive(csp, new_key, new_values)
+    dirs = get_directives(csp)
+    new_dirs = get_directives(new_csp)
+    assert new_key in new_dirs
+    for key in new_dirs:
+        if key != new_key:
+            assert key in dirs and new_dirs[key] == dirs[key]
+        else:
+            assert new_dirs[key] == dirs.get(key, set()).union(set(new_values))
+    return new_dirs
```

Comment:
This is too complicated for me to understand.

Can we construct a CSP as a simple string, run the `csp_add_directive` function, and assert the result is equal to some string literal.